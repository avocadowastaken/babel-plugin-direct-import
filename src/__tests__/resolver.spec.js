const { resolveFilename } = require("../resolver");

describe("resolver", () => {
  describe("resolveFilename", () => {
    it("should resolve package path", () => {
      const filePath = resolveFilename("lodash");

      expect(filePath).toContain("node_modules/lodash/lodash.js");
    });

    it("should throw errors from `Module` package", () => {
      expect(() => resolveFilename(null)).toThrowErrorMatchingSnapshot();
    });

    it("should not throw when module is not installed", () => {
      const fn = jest.fn();
      const warn = console.warn;

      console.warn = fn;

      const filePath = resolveFilename("foo");

      expect(filePath).toBeNull();
      expect(fn.mock.calls).toMatchSnapshot();

      console.warn = warn;
    });
  });
});
