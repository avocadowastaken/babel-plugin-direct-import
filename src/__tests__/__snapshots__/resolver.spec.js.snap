// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolver resolveFilename should not throw when module is not installed 1`] = `
Array [
  Array [
    "babel-plugin-direct-import: Cannot find module 'foo'",
  ],
]
`;

exports[`resolver resolveFilename should throw errors from \`Module\` package 1`] = `"Cannot read property 'length' of null"`;
